tardis/plugins/auditor.py

```diff
@@ -0,0 +1,115 @@
+from ..configuration.configuration import Configuration
+from ..interfaces.plugin import Plugin
+from ..interfaces.state import State
+from ..utilities.attributedict import AttributeDict
+from ..resources.dronestates import AvailableState, DownState
+
+import pyauditor
+
+import logging
+import pytz
+from tzlocal import get_localzone
+
+
+class Auditor(Plugin):
+    """
+    The :py:class:`~tardis.plugins.auditor.Auditor` plugin is a collector for the
+    accounting tool Auditor. It sends accounting information of individual drones to an
+    Auditor instance. The records contain information about the provided resources of
+    the drones as well as start and stop times. When a drone enters `AvailableState`, a
+    record with the start time set to the time it went into this state is stored in the
+    Auditor database. The stop time remains empty until the drone goes into `DownState`.
+    The Auditor plugin does not keep any state.
+    """
+
+    def __init__(self):
+        self.logger = logging.getLogger("cobald.runtime.tardis.plugins.auditor")
+        config = Configuration()
+        config_auditor = config.Plugins.Auditor
+
+        self._resources = {}
+        self._components = {}
+        for site in config.Sites:
+            self._resources[site.name] = {}
+            self._components[site.name] = {}
+            for machine_type in getattr(config, site.name).MachineTypes:
+                self._resources[site.name][machine_type] = {}
+                self._components[site.name][machine_type] = {}
+                for r in getattr(config, site.name).MachineMetaData[machine_type]:
+                    self._resources[site.name][machine_type][r] = getattr(
+                        config, site.name
+                    ).MachineMetaData[machine_type][r]
+                    self._components[site.name][machine_type][r] = getattr(
+                        config_auditor.components, machine_type
+                    ).get(r, {})
+
+        self._user = config_auditor.user if config_auditor.user else "tardis"
```

Comment:
Accessing `config_auditor.user` should raise an `AttributeError` if it does not exists.

Reply:
Would propose `getattr(config_auditor, "user", "tardis")` instead.